src/main/java/mongodb/Main.java

```diff
@@ -0,0 +1,240 @@
+        return;
+      }
+    } catch (MongoException me) {
+      System.err.println("Unable to find a recipe to update in MongoDB due to an error: " + me);
+      return;
+    }
+
+    /*      *** UPDATE A DOCUMENT ***
+     *
+     * You can update a single document or multiple documents in a single call.
+     *
+     * Here we update the PrepTimeInMinutes value on the document we
+     * just found.
+     */
+    Bson updateFilter = Updates.set("prepTimeInMinutes", 72);
+
+    // The following FindOneAndUpdateOptions specify that we want it to return
+    // the *updated* document to us. By default, we get the document as it was *before*
+    // the update.
+    FindOneAndUpdateOptions options = new FindOneAndUpdateOptions().returnDocument(ReturnDocument.AFTER);
+
+    // The updatedDocument object is a Recipe object that reflects the
+    // changes we just made.
+    try {
+      Recipe updatedDocument = collection.findOneAndUpdate(findPotato,
+              updateFilter, options);
+      if (updatedDocument == null) {
+        System.out.println("Couldn't update the recipe. Did someone (or something) delete it?");
+      } else {
+        System.out.println("Updated the recipe to: " + updatedDocument);
+      }
+    } catch (MongoException me) {
+      System.err.println("Unable to update any recipes due to an error: " + me);
+    }
+
+    /*      *** DELETE DOCUMENTS ***
+     *
+     *      As with other CRUD methods, you can delete a single document
+     *      or all documents that match a specified filter. To delete all
+     *      of the documents in a collection, pass an empty filter to
+     *      the deleteMany() method. In this example, we'll delete 2 of
+     *      the recipes.
+     */
+    Bson deleteFilter = Filters.in("name", Arrays.asList("elotes", "fried rice"));
+    try {
+      DeleteResult deleteResult = collection
+              .deleteMany(deleteFilter);
+      System.out.println(String.format("Deleted %d documents.", deleteResult.getDeletedCount()));
+    } catch (MongoException me) {
+      System.err.println("Unable to delete any recipes due to an error: " + me);
+    }
+
+    // always close the connection when done working with the client
+    mongoClient.close();
+  }
+
+  // POJO (Plain Old Java Object) class defining a recipe. This class is a POJO because it contains getters and
+  // setters for every member variable as well as an empty constructor.
+  public static class Recipe {
+    private String name;
+    private List<String> ingredients;
+    private int prepTimeInMinutes;
+
+    public Recipe(String name, List<String> ingredients, int prepTimeInMinutes) {
+      this.name = name;
+      this.ingredients = ingredients;
+      this.prepTimeInMinutes = prepTimeInMinutes;
+    }
+
+    // empty constructor required when we fetch data from the database -- getters and setters are later used to
+    // set values for member variables
+    public Recipe() {
```

Comment:
I was reading more about the PojoCodecProvider since it wasn't clear to me why you need the empty constructor. Have you tested this code without the initialization contained here? I think the codec should work without them based on this statement: 
> By default a POJO Codec works with POJOs that have an empty, no arguments, constructor. 

However, if you initialize the fields, when converting to BSON during an insert or update, it will add whatever you've initialized them to into the document. E.g. if you initialize ingredients to an empty list and then insert the document, the resulting document in MongoDB will contain "ingredients: []" in that document. If you do not initialize ingredients, when you insert that document to MongoDB, the resulting document will not have an "ingredients" field.

Reply:
It is required, and removing the constructor produces the following error:

```
Exception in thread "main" org.bson.codecs.configuration.CodecConfigurationException: An exception occurred when decoding using the AutomaticPojoCodec.
Decoding into a 'Recipe' failed with the following exception:

Cannot find a public constructor for 'Recipe'.

A custom Codec or PojoCodec may need to be explicitly configured and registered to handle this type.
...
```

Reply:
Personally I think it's better to have an empty field guaranteed as default in this case, though I'm sure there are situations where you might not want that behavior.

Reply:
I know it is required -- i was asking about removal of the initialization not the constructor.